containerd-shim-v2/exec.go

```diff
@@ -73,9 +73,11 @@ func newExec(c *container, stdin, stdout, stderr string, terminal bool, jspec *g
 
 	// process exec request
 	var spec specs.Process
-	if err := json.Unmarshal(jspec.Value, &spec); err != nil {
+	v, err := typeurl.UnmarshalAny(jspec)
+	if err != nil {
 		return nil, err
 	}
+	spec = *v.(*specs.Process)
```

Comment:
Please add a switch case to check for the type, and if this is not the expected type, let's throw an error.

Reply:
@sboeuf updated.